shimmingtoolbox/load_nifti.py

```diff
@@ -0,0 +1,168 @@
+
+            select_run = int(input_resp)
+
+            if select_run in list(run_list.keys()):
+                break
+            else:
+                logging.error(f"Input must be linked to a run number. {input_resp} is out of range")
+    else:
+        select_run = list(run_list.keys())[0]
+        logging.info(f"Reading acquisitions for run {list(run_list.keys())[0]}")
+
+    # Create output array and headers
+    nifti_pos = 0
+    echo_shape = sum(1 for tmp_info in run_list[select_run] if modality in tmp_info[1])
+    if nifti_init[0][0].ndim == 3:
+        niftis = np.empty([nifti_init[0][0].shape[0], nifti_init[0][0].shape[1], nifti_init[0][0].shape[2], echo_shape, 1], dtype=float)
+        for i_echo in range(len(run_list[select_run])):
+            tmp_nii = run_list[select_run][i_echo][0]
+            if modality in run_list[select_run][i_echo][1]:
+                info.append(tmp_nii[0].header)
+                json_info.append(tmp_nii[1])
+                niftis[:, :, :, nifti_pos, 0] = tmp_nii[2]
+                nifti_pos += 1
+    else:
+        niftis = np.empty([
+            nifti_init[0][0].shape[0], nifti_init[0][0].shape[1], nifti_init[0][0].shape[2],
+            echo_shape, nifti_init[0][0].shape[3]], dtype=float)
+        for i_echo in range(len(run_list[select_run])):
+            if modality in run_list[select_run][i_echo][1]:
+                tmp_nii = run_list[select_run][i_echo][0]
+                info.append(tmp_nii[0].header)
+                json_info.append(tmp_nii[1])
+                for i_volume in range(nifti_init[0][0].shape[3]):
+                    niftis[:, :, :, nifti_pos, i_volume] = tmp_nii[2][:, :, :, i_volume]
+                nifti_pos += 1
+
+    return niftis, info, json_info
+
+
+def read_nii(nii_path, auto_scale=True):
+    """ Reads a nifti file and returns the corresponding image and info. Also returns the associated json data.
+    Args:
+        nii_path (str): direct path to the .nii or .nii.gz file that is going to be read
+        auto_scale (:obj:`bool`, optional): Tells if scaling is done before return
+    Returns:
+        info (Nifti1Image): Objet containing various data about the nifti file (returned by nibabel.load)
+        json_data (dict): Contains the different fields present in the json file corresponding to the nifti file
+        image (ndarray): Image contained in the read nifti file. Siemens phase images are rescaled between 0 and 2pi.
+    """
+
+    info = nib.load(nii_path)
+
+    # `extractBefore` should get the correct filename in both.nii and.nii.gz cases
+    json_path = nii_path.split('.nii')[0] + '.json'
+
+    if os.path.isfile(json_path):
+        json_data = json.load(open(json_path))
+    else:
+        raise ValueError('Missing json file')
+
+    image = np.asarray(info.dataobj)
+    if auto_scale:
+        if ('Manufacturer' in json_data) and (json_data['Manufacturer'] == 'Siemens') \
+                and ("phase" in json_data['ImageComments']):
```

Comment:
This seems like an edge case. "ImageComments" won't always have in the description if it is phase data or not. For example, our testing data does not even have the field ImageComments and uses the more useful ImageType to differentiate between phase, mag and complex.

Reply:
It would also maybe be better to check for "*phase*" instead of just "phase" in case the user decides to write some other comments in the "ImageComments" tag.

Reply:
I was checking the ImageType before, but this field is also not always present in the json file outputted by dcm2bids. I read the other issues to find that @gaspardcereza was adding the phase, mag, complex tag in the ImageComments. That's why I used it.